src/muz/spacer/spacer_convex_closure.cpp

```diff
@@ -278,19 +278,23 @@ bool convex_closure::closure(expr_ref_vector &res_vec) {
 
     if (red_dim > m_st.m_max_dim) m_st.m_max_dim = red_dim;
     if (red_dim > 1) {
-        SASSERT(m_nw_vars.size() == 0);
-        TRACE("subsume", tout << "Computing syntactic convex closure\n";);
-        // TODO: add an option to disable syn cls and use it for bv
-        syn_cls(res_vec);
-        return false;
+        // there is no alternative to syntactic convex closure right now
+        // syntactic convex closure does not support BV
+        if (m_do_syn_cls) {
+            SASSERT(m_nw_vars.size() == 0);
+            TRACE("subsume", tout << "Computing syntactic convex closure\n";);
+            syn_cls(res_vec);
+        }
+        else NOT_IMPLEMENTED_YET();
+        return true;
     }
 
     // zero dimensional convex closure
-    if (red_dim == 0) { return true; }
+    if (red_dim == 0) { return false; }
 
     SASSERT(red_dim == 1);
     do_one_dim_cls(var, res_vec);
-    return true;
+    return false;
```

Comment:
why `false`? Should not `do_one_dim_cls` decide whether it can compute an exact closure or not